src/main/java/com/prowidesoftware/swift/utils/IsoUtils.java

```diff
@@ -51,21 +54,15 @@ private IsoUtils() {
         log.fine("IsoUtils initialized with " + currencies.size() + " currency codes and " + countries.size() + " country codes");
     }
 
-    public static synchronized IsoUtils getInstance() {
-        if (instance == null) {
-            synchronized (IsoUtils.class) {
-                if (instance == null) {
-                    instance = new IsoUtils();
-                }
-            }
-        }
-        return instance;
+    public static IsoUtils getInstance() {
+        return INSTANCE;
     }
 
     public Set<String> getCurrencies() {
         return currencies;
     }
 
+    @ProwideDeprecated(phase2 = TargetYear.SRU2022, comment = "singleton should be immutable, method can be package protected for testing")
```

Comment:
I don't agree on this one. A singleton can be mutable or immutable. And in this case is quite useful because the default set could be missing entries, or could have entries that have to be excluded. Maybe the the class should not be a singleton at all, but being a singleton I don't see why it cannot be mutable.